modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -180,13 +179,13 @@ void updateTaskState() {
     }
 
     //visible for testing
-    int indexChunks(String name, InputStream is, int chunk, String expectedMd5) throws IOException {
+    int indexChunks(String name, InputStream is, int chunk, String expectedMd5, long timestamp) throws IOException {
         MessageDigest md = MessageDigests.md5();
         for (byte[] buf = getChunk(is); buf.length != 0; buf = getChunk(is)) {
             md.update(buf);
-            client.prepareIndex(DATABASES_INDEX).setId(name + "_" + chunk)
```

Comment:
Maybe keep the _id but with timestamp? That way the _id has meaning and if due to some issue we index a document with the same _id then we fail with an error (b/c create=true).

Reply:
I switched back to using _id with added timestamp as you suggested